src/Admin/Polyfills.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * Handles backward compatibility of assets for older versions of WP.
+ *
+ * @since 2.0
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP\Admin;
+
+use AmpProject\AmpWP\Infrastructure\Delayed;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use WP_Scripts;
+use WP_Styles;
+
+/**
+ * Registers assets that may not be available in the current site's version of core.
+ *
+ * @since 2.0
+ */
+final class Polyfills implements Delayed, Service, Registerable {
+
+	/**
+	 * Get the action to use for registering the service.
+	 *
+	 * @return string Registration action to use.
+	 */
+	public static function get_registration_action() {
+		return 'amp_register_polyfills';
+	}
+
+	/**
+	 * Runs on instantiation.
+	 */
+	public function register() {
+		$this->register_shimmed_scripts( wp_scripts() );
+		$this->register_shimmed_styles( wp_styles() );
```

Comment:
Per above, then this could be done:

```suggestion
		add_action( 'wp_default_scripts', [ $this, 'register_shimmed_scripts' ] );
		add_action( 'wp_default_styles', [ $this, 'register_shimmed_styles' ] );
```

Reply:
I did try this at first, but `wp_default_styles` runs too early for the contextual information we need to register the polyfills conditionally.